### PR TITLE
Reduce the number of calculating usage

### PR DESCRIPTION
**Pull request by FengyunPan**
The quotaEvaluator call Usage() to calculate usage twice for every
admission attributes, and calculating usage will waste much time.

### CODE REVIEW

**Review by derekwaynecarr — CHANGES_REQUESTED** [review]
if we move the code as noted, i think we need to make sure the block here is also moved above.

https://github.com/kubernetes/kubernetes/pull/44149/files#diff-1622c871bcd912b43639c9d3d441e713L445

**Comment by FengyunPan** [comment]
@derekwaynecarr Alright, I will update it soon.

**Comment by FengyunPan** [comment]
@derekwaynecarr Thank for your review, I have updated it.

**Comment by derekwaynecarr** [comment]
@k8s-bot ok to test

File: plugin/pkg/admission/resourcequota/controller.go [hunk 1]
```diff
@@ -379,17 +379,28 @@ func (e *quotaEvaluator) checkRequest(quotas []api.ResourceQuota, a admission.At
...
+	// the resource represents a number of unique references to external
+	// resource. In such a case an evaluator needs to process other objects in
+	// the same namespace which needs to be known.
+	if accessor, err := meta.Accessor(inputObject); namespace != "" && err == nil {
+		if accessor.GetNamespace() == "" {
+			accessor.SetNamespace(namespace)
+		}
+	}
+
+	// if we have limited resources enabled for this resource, always calculate usage
```
Comment (derekwaynecarr) [inline plugin/pkg/admission/resourcequota/controller.go]: this issue i have with the code now is that usage is calculated all the time.  we should only need to calculate use if there is a matching limited resource (which is an alpha feature at moment) or if there is an actually interesting covering quota.  can this be structured to ensure that usage is only calculated if the above constraints are satisfied?

**Review by derekwaynecarr — CHANGES_REQUESTED** [review]
this appears to calculate usage even if there is no interesting quota or limited resource.  we should only calculate usage if there is an interesting quota or a limited resource.

**Pull request closed without merging**